src/plugins/query_enhancements/public/query_assist/components/query_assist_bar.tsx

```diff
@@ -37,19 +34,9 @@ export const QueryAssistBar: React.FC<QueryAssistInputProps> = (props) => {
   const { generateQuery, loading } = useGenerateQuery();
   const [callOutType, setCallOutType] = useState<QueryAssistCallOutType>();
   const dismissCallout = () => setCallOutType(undefined);
-  const [selectedIndex, setSelectedIndex] = useState<string>('');
-  const dataSourceIdRef = useRef<string>();
+  const selectedIndex = services.data.query.dataSet.getDataSet()?.title;
```

Comment:
Will the component re-render if the dataset changes? If not you should subscribe to the dataset. use the useDataSetManager hook to get a stateful dataset object

Reply:
i agree. `dataSourceIdRef` previously was a ref and we never relied on re-rendering triggered by subscription, so current implementation shouldn't cause issues. But the re-rendering is probably a side effect and not good to rely on.

i wanted to subscribe but i didn't understand this part https://github.com/opensearch-project/OpenSearch-Dashboards/pull/7464#discussion_r1690533991

it doesn't work unless i add a default initial value